content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+
+Referential actions are **not currently** supported on [implicit many-many relations](./relations#implicit-vs-explicit-many-to-many-relations), instead you must create a [join table](./relations#how-to-use-a-relation-table-with-a-many-to-many-relationship) in your schema file and explicitly set the referential actions on the many-many relation.
+
+</Admonition>
+
+## Types of referential actions
+
+### Cascade
+
+When used with `onDelete`, The `Cascade` action deletes all records that are dependant on the record being deleted. 
+
+When used with `onUpdate`, it updates the relation scalar filed if the referenced scalar fields of the dependant record are updated.
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade, onUpdate: Cascade)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+<Admonition type="info">
+
+`onUpdate` defaults to `Cascade` when not set. This applies to both **optional** and **required** relations. `onDelete` defaults to `Cascade` on **required** relations.
+
+</Admonition>
+
+##### Result of using `Cascade`?
+
+If a `User` record is deleted then their posts are deleted too. If a users id is updated then the corresponding `authorId` is also updated
+
+### Restrict
+
+The `Restrict` action will prevent the update or deletion of any parent records which have connected children. 
+
+> **Note**: The children records can still be deleted
+
+#### Example usage
+
+```prisma file=schema.prisma highlight=5;add|4;delete
+model Post {
+    id        Int     @id @default(autoincrement())
+    title     String
+    author    User    @relation(fields: [authorId], references: [id])
+    author    User    @relation(fields: [authorId], references: [id], onDelete: Restrict)
+    authorId  Int
+}
+
+model User {
+  id    Int         @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `Restrict`?
+
+If a `User` has a related post in the `Post` table you will **not** be able to delete the `User`, you will be able to delete (all) the posts however.
+
+<Admonition type="warning">
+
+The `Restrict` action is not is **not** available on [SQL Server](../../database-connectors/microsoft-sql-server) and will trigger a schema validation error. Instead, use the equivalent [`NoAction`](#noaction).
+
+</Admonition>
+
+### NoAction
+
+The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependant on the database being used. But in general they both have the same functionality. In Postgres however, `NoAction` defers the check until later in the transaction.
```

Comment:
```suggestion
The `NoAction` action is similar to `Restrict` in that it will prevent the update or deletion of any records which are connected and throw an error. The difference between the two is dependent on the database being used: In Postgres, `NoAction` defers the check until later in the transaction. TODO MySQL. TODO SQL Server. Besides these details, both `NoAction` and `Restrict` offer the same functionality. 
```

Reply:
Waiting on feedback in slack for help with this